drivers/hmis/spec/models/hmis/form/processor_spec.rb

```diff
@@ -0,0 +1,125 @@
+require 'rails_helper'
+
+RSpec.describe Hmis::Form::Processor, type: :model do
+  let!(:ds) { create :hmis_data_source }
+  let!(:user) { create(:user).tap { |u| u.add_viewable(ds) } }
+  let(:hmis_user) { Hmis::User.find(user.id)&.tap { |u| u.update(hmis_data_source_id: ds.id) } }
+  let(:hmis_hud_user) { Hmis::Hud::User.from_user(hmis_user) }
+
+  let!(:fd) { create :hmis_form_definition }
+
+  let(:o1) { create :hmis_hud_organization, data_source: ds, user: hmis_hud_user }
+  let(:p1) { create :hmis_hud_project, data_source: ds, organization: o1, user: hmis_hud_user }
+  let(:c1) { create :hmis_hud_client, data_source: ds, user: hmis_hud_user }
+  let!(:e1) { create :hmis_hud_enrollment, data_source: ds, project: p1, client: c1, user: hmis_hud_user }
+
+  it 'ingests EnrollmentCoC into the hud tables' do
+    assessment = Hmis::Hud::Assessment.new_with_defaults(enrollment: e1, user: hmis_hud_user, form_definition: fd, assessment_date: Date.current)
+    assessment.assessment_detail.hud_values = {
+      'EnrollmentCoc.cocCode' => 'MA-507',
+    }
+
+    assessment.assessment_detail.processor.run!
+    assessment.save_not_in_progress
+
+    expect(assessment.enrollment.enrollment_cocs.count).to eq(1)
+    expect(assessment.enrollment.enrollment_cocs.first.coc_code).to eq('MA-507')
+  end
+
+  it 'ingests IncomeBenefit into the hud tables' do
+    assessment = Hmis::Hud::Assessment.new_with_defaults(enrollment: e1, user: hmis_hud_user, form_definition: fd, assessment_date: Date.current)
+    assessment.assessment_detail.hud_values = {
+      'IncomeBenefit.incomeFromAnySource' => 'YES',
+      'IncomeBenefit.earned' => true,
+      'IncomeBenefit.earnedAmount' => 1,
+      'IncomeBenefit.otherInsurance' => false,
+    }
+
+    assessment.assessment_detail.processor.run!
+    assessment.save_not_in_progress
+
+    expect(assessment.enrollment.income_benefits.count).to eq(1)
+
+    income_benefits = assessment.enrollment.income_benefits.first
+    expect(income_benefits.income_from_any_source).to eq(1)
+    expect(income_benefits.earned).to eq(1)
```

Comment:
This is correct behavior (`true` saved as `1`) but how is it working? I don't understand because the GraphQL Schema type is Boolean and I'm not seeing a transformation.

Can you add test cases to cover that `false` gets saved as `0` and `nil` gets saved as `99`?

Reply:
These rely on the AR database translations, which seem to work for the cases we have, so I didn't want to complicate the code/

There is `false` in this test (`'IncomeBenefit.otherInsurance' => false`), and I think we decided the `nil` is clearing the field, so you will need to send a `99` if you want to store one. Is this incorrect?